server/etcdserver/api/membership/cluster.go

```diff
@@ -542,6 +543,53 @@ func (c *RaftCluster) PromoteMember(id types.ID, shouldApplyV3 ShouldApplyV3) {
 	)
 }
 
+func (c *RaftCluster) SyncLearnerPromotionIfNeeded() {
+	c.Lock()
+	defer c.Unlock()
+
+	v2Members, _ := membersFromStore(c.lg, c.v2store)
+	v3Members, _ := membersFromBackend(c.lg, c.be)
+
+	for id, v2Member := range v2Members {
+		v3Member, ok := v3Members[id]
+
+		if !ok {
+			c.lg.Error("Detected member only in v2store but missing in v3store", zap.String("member", fmt.Sprintf("%+v", *v2Member)))
+			continue
+		}
+
+		// A peerURL list is considered the same regardless of order.
+		clonedV2Member := v2Member.Clone()
+		sort.Strings(clonedV2Member.PeerURLs)
+
+		clonedV3Member := v3Member.Clone()
+		sort.Strings(clonedV3Member.PeerURLs)
+
+		if reflect.DeepEqual(clonedV2Member.RaftAttributes, clonedV3Member.RaftAttributes) {
```

Comment:
Could you just compare IsLearner? We don't care about other information

Reply:
RaftAttribute includes both peerURLs and IsLearner, both of them should always match between v2 and v3store. I think it's more prudent to double check peerURL matches.

Reply:
can we resolve this comment?